Update GitHub workflows

flavio opened the pull request:
Use the new reusable workflows and GitHub Actions provided by the
Kubewarden project.

Also, improve the Makefile to have better targets.

flavio commented:
Converted to be a draft, cargo-generate is not working as expected...

flavio commented:
Everything is working now, this is ready to be reviewed

jvanz commented on Makefile:
```diff
@@ -10,10 +15,15 @@ fmt:
 
+.PHONY: e2e-tests
+e2e-tests: annotated-policy.wasm
+	bats e2e.bats
```
Should not we add a fake test file to avoid errors right after the users create their projects?

jvanz commented on Makefile:
```diff
@@ -10,10 +15,15 @@ fmt:
 
+.PHONY: e2e-tests
+e2e-tests: annotated-policy.wasm
+	bats e2e.bats
```
Should not we add a fake test file to avoid errors right after the users create their projects?

flavio replied:
This is correct! Let me add something

flavio commented:
This can be reviewed again.

Changes since the last time are inside of 2 dedicated commits:
  * ensure all the files are rendered correctly at generation time. Some files had the template placeholders not processed at all
  * provide a simple e2e test that works out of the box

viccuad commented on .github/workflows/release.yml:
```diff
@@ -0,0 +1,33 @@
+    needs: test
+    permissions:
+      # Required to push to GHCR
+      contents: write
```
the current comment is somewhat incorrect, as `contents` is required for creating GH Releases. I would prefer something like this:
```
      # Required to create GH releases
      contents: write
      # Required to push to GHCR
      packages: write
      # Required by cosign keyless signing
      id-token: write
```

flavio replied:
fixed

viccuad commented on .gitignore:
```diff
@@ -1,4 +1,6 @@
 /target
+policy.wasm
```
Could be simplified to `*.wasm`

flavio replied:
done

viccuad reviewed (approved):
LGTM, with nitpicks

flavio commented:
ready to be reviewed again
